sql/hive/src/main/scala/org/apache/spark/sql/hive/hiveUdfs.scala

```diff
@@ -179,7 +179,12 @@ private[hive] case class HiveGenericUdf(funcWrapper: HiveFunctionWrapper, childr
         })
       i += 1
     }
-    unwrap(function.evaluate(deferedObjects), returnInspector)
+    
+    if (function.getUdfName().endsWith("UDFCurrentDB")) {
```

Comment:
Can you explain why you think returning a `null` is more reasonable than executing the `UDFCurrentDB`?  Seems it will not throws exception anymore in Hive 0.14: http://grepcode.com/file/repo1.maven.org/maven2/org.apache.hive/hive-exec/0.14.0/org/apache/hadoop/hive/ql/udf/generic/UDFCurrentDB.java/